honeybee_doe2/writer.py

```diff
@@ -21,7 +21,12 @@ def model_to_inp(hb_model):
     comp_data = ComplianceData()
     sb_data = sbd()
 
-    hb_model.convert_to_units(units='Feet')
+    hb_model = hb_model.duplicate()
+
+    if hb_model.units != 'Feet':
+        hb_model.convert_to_units(units='Feet')
+    hb_model.remove_degenerate_geometry()
```

Comment:
FYI, this will be removing degenerate objects at the model tolerance if you aren't passing a specific value to the method. This should get coordinated with all of the other tolerance values in the package to make sure that all of the methods succeed.